pkg/osquery/tables/execparsers/rpm/parser.go

```diff
@@ -0,0 +1,60 @@
+package rpm
+
+import (
+	"bufio"
+	"io"
+	"strings"
+
+	"golang.org/x/exp/slices"
+)
+
+var allowedKeys = []string{
+	"name",
+	"version",
+	"release",
+	"install date",
+	"group",
+	"build date",
+	"summary",
+	"description",
+}
+
+func rpmParse(reader io.Reader) (any, error) {
+	results := make([]map[string]string, 0)
+	row := make(map[string]string)
+	readingDesc := false
+
+	scanner := bufio.NewScanner(reader)
+	for scanner.Scan() {
+		line := scanner.Text()
+		// We expect rpm to return lines in the following format:
+		// `Name: pm-utils`
+		// `Version: 1.4.1`
+		// `Release: 27.el7`...
+		// We split each line to get a key -> value pair, then store
+		// it into our row until we log the row and start a new one.
+		kv := strings.SplitN(line, ":", 2)
+		var key = strings.ToLower(strings.TrimSpace(kv[0]))
+		if slices.Contains(allowedKeys, key) {
+			// rpm doesn't provide a clean break. I'm collecting everything until
+			// description, then storing each line for description until the next key.
+			if key == "description" {
+				readingDesc = true
+			} else if readingDesc {
+				readingDesc = false
+				results = append(results, row)
+				row = make(map[string]string)
```

Comment:
I'm confused by this. Because it sets `readingDesc = false` the next loop shouldn't come back here, so how does it handle the multiline descriptions?

Reply:
It won't fall into this section until a key is matched to the allowedKeys, so when the multiline description is first found readingDesc gets set to true, and until another allowedKey is found it'll append each line to the description. When the next key i.e. name is found, it sets readingDesc false and stores the previous row before making a new one and storing name into it.

Reply:
Oh... So it only works because Name and Description are both in allowedkeys

Reply:
Might need more comments, but the data is messy so the parser is also messy. And there's a solid test case.